Import necessary libraries

In [77]:
import os
import json
import pandas as pd

Code to basic functionalities used later on in the code

In [79]:
def load_json(file_name):
    with open(file_name, encoding='utf-8') as f:
        iter_f = iter(f)
        line = f.readline()
        list_name = []
        for line in iter_f:
            d = json.loads(line)
            list_name.append(d)
        f.close()
    return pd.DataFrame(list_name)


Change directory to folder with the JSON files from "Yelp Data"

In [80]:
print(os.getcwd())
os.chdir("/Users/ashvini/Desktop/Fall2020/Stat628/Module3/Data")

/Users/ashvini/Desktop/Fall2020/Stat628/Module3/Data


Load JSON files as dataframes

In [45]:
business = load_json('business_city.json')
review = load_json('review_city.json')

In [81]:
business.columns

Index(['business_id', 'name', 'address', 'city', 'state', 'postal_code',
       'latitude', 'longitude', 'stars', 'review_count', 'is_open',
       'attributes', 'categories', 'hours'],
      dtype='object')

In [82]:
business = business[business.is_open == 1]
state_counts = business.groupby('state').count()
state_counts['business_id'].sort_values(ascending = False).head(10)

state
OH    13762
PA    10206
WI     4433
IL     1592
Name: business_id, dtype: int64

As we can see based on geography, 3 of top 10 states are Ohio, Pennsylvania, and Wisconsin. We will be focuisng on these these states from the United States.

In [83]:
state_list = ['OH','PA','WI']
business1 = business[business.state.isin(state_list)]
business1.shape

(28401, 14)

In [84]:
category_count = business1.groupby('categories').count()
category_count['state'].sort_values(ascending = False).head(20)

categories
Restaurants, Pizza                    239
Pizza, Restaurants                    237
Beauty & Spas, Nail Salons            205
Nail Salons, Beauty & Spas            184
Restaurants, Chinese                  134
Chinese, Restaurants                  116
Beauty & Spas, Hair Salons            105
Food, Coffee & Tea                     94
Hair Salons, Beauty & Spas             94
Grocery, Food                          93
Auto Repair, Automotive                91
Pets, Veterinarians                    88
Coffee & Tea, Food                     88
Food, Grocery                          87
Sandwiches, Restaurants, Fast Food     83
Automotive, Auto Repair                83
Restaurants, Sandwiches, Fast Food     78
Veterinarians, Pets                    75
Beauty & Spas, Barbers                 72
Food, Ice Cream & Frozen Yogurt        72
Name: state, dtype: int64

Pizza is the most favored category
We therefore focus on the restaurants which serve pizza.

In [85]:
business2 = business1[business1['categories'].str.contains('pizza',case = False).fillna(False)]
business2.shape

(1505, 14)

In [86]:
business.columns

Index(['business_id', 'name', 'address', 'city', 'state', 'postal_code',
       'latitude', 'longitude', 'stars', 'review_count', 'is_open',
       'attributes', 'categories', 'hours'],
      dtype='object')

In [87]:
review.columns

Index(['review_id', 'user_id', 'business_id', 'stars', 'useful', 'funny',
       'cool', 'text', 'date'],
      dtype='object')

In [88]:
US = pd.merge(business2[business2.review_count >= 10] , review , on = 'business_id')
US.shape

(62091, 22)

In [89]:
US.groupby('state').business_id.nunique().sum()

1053

    State: OH,PA,WI
    Category: 'pizza' 

There are 1053 restaurants with a review count of 62091.